src/poly/mod.rs

```diff
@@ -1,7 +1,7 @@
 use std::{
     collections::HashMap,
     fmt::Debug,
-    hash::Hash,
+    hash::{DefaultHasher, Hash, Hasher},
```

Comment:
Seems like Hasher is not used.

Reply:
Hasher is a trait that implements `finish` that is being used from `DefaultHasher`

Reply:
Got it!